### ISSUE TITLE
HelloCompute running failed under Win10.

### ISSUE THREAD

**Issue by lygstate** [google/nxt-standalone#148]
E:\CI-Cor-Ready\ai\face-demo-work\nxt-standalone\build\Debug>HelloCompute.exe
Assertion failure at E:\CI-Cor-Ready\ai\face-demo-work\nxt-standalone\src\utils\D3D12Binding.cpp:49 (ASSERT_SUCCESS): (((HRESULT)(hr)) >= 0)

**Comment by kainino0x** [comment]
Do the non-compute demos have this issue?

I don't know why this would happen without being able to debug it further. But maybe it could be that your hardware does not support D3D12 compute. (Not sure if that's possible.)

**Comment by lygstate** [comment]
@kainino0x Not only compute, Triangle also failed. Only appeared in Debug version, Release version are OK.

**Comment by kainino0x** [comment]
That would be because we don't process ASSERTs in the Release build.

Do the demos work in release despite that issue? If so, you can just comment out that ASSERT if you really need it to run.

If you send a full stack trace for the assert failure we might be able to give more advice.

**Comment by lygstate** [comment]
···
>	HelloCompute.exe!HandleAssertionFailure(const char * file, const char * function, int line, const char * condition) Line 25	C++
 	HelloCompute.exe!utils::`anonymous namespace'::ASSERT_SUCCESS(HRESULT hr) Line 49	C++
 	HelloCompute.exe!utils::SwapChainImplD3D12::Present() Line 232	C++
 	HelloCompute.exe!utils::SwapChainImpl::GenerateSwapChainImplementation::__l2::<lambda>(void * userData) Line 39	C++
 	HelloCompute.exe!const char * <lambda>(void *)::<lambda_invoker_cdecl>(void * userData) Line 39	C++
 	HelloCompute.exe!backend::SwapChainBase::Present(backend::TextureBase * texture) Line 85	C++
 	HelloCompute.exe!backend::d3d12::`anonymous namespace'::ValidatingSwapChainPresent(backend::d3d12::SwapChain * self, backend::d3d12::Texture * texture) Line 7939	C++
 	HelloCompute.exe!nxt::wire::server::Server::HandleSwapChainPresent(const unsigned char * * commands, unsigned __int64 * size) Line 7791	C++
 	HelloCompute.exe!nxt::wire::server::Server::HandleCommands(const unsigned char * commands, unsigned __int64 size) Line 1142	C++
 	HelloCompute.exe!nxt::wire::TerribleCommandBuffer::Flush() Line 47	C++
 	HelloCompute.exe!DoFlush() Line 224	C++
 	HelloCompute.exe!frame() Line 161	C++
 	HelloCompute.exe!main(int argc, const char * * argv) Line 172	C++

···

**Comment by lygstate** [comment]
```
>	CppHelloTriangle.exe!HandleAssertionFailure(const char * file, const char * function, int line, const char * condition) Line 25	C++
 	CppHelloTriangle.exe!utils::`anonymous namespace'::ASSERT_SUCCESS(HRESULT hr) Line 49	C++
 	CppHelloTriangle.exe!utils::SwapChainImplD3D12::Present() Line 232	C++
 	CppHelloTriangle.exe!utils::SwapChainImpl::GenerateSwapChainImplementation::__l2::<lambda>(void * userData) Line 39	C++
 	CppHelloTriangle.exe!const char * <lambda>(void *)::<lambda_invoker_cdecl>(void * userData) Line 39	C++
 	CppHelloTriangle.exe!backend::SwapChainBase::Present(backend::TextureBase * texture) Line 85	C++
 	CppHelloTriangle.exe!backend::d3d12::`anonymous namespace'::ValidatingSwapChainPresent(backend::d3d12::SwapChain * self, backend::d3d12::Texture * texture) Line 7939	C++
 	CppHelloTriangle.exe!nxt::wire::server::Server::HandleSwapChainPresent(const unsigned char * * commands, unsigned __int64 * size) Line 7791	C++
 	CppHelloTriangle.exe!nxt::wire::server::Server::HandleCommands(const unsigned char * commands, unsigned __int64 size) Line 1142	C++
 	CppHelloTriangle.exe!nxt::wire::TerribleCommandBuffer::Flush() Line 47	C++
 	CppHelloTriangle.exe!DoFlush() Line 224	C++
 	CppHelloTriangle.exe!frame() Line 172	C++
 	CppHelloTriangle.exe!main(int argc, const char * * argv) Line 183	C++
 	[External Code]	

···

**Comment by Kangz** [comment]
I think there's an issue with the D3D12 swapchain code: tests work without issues but examples all fail on my home desktop too.

@lygstate try running examples with "-b opengl" for now.

**Comment by lygstate** [comment]
Open failed more earlier on my computer

**Comment by msiglreith** [comment]
Seems to be working now.

**Comment by Kangz** [comment]
Yep, fixed it in #158
